sql/src/test/java/org/apache/druid/sql/calcite/CalciteParameterQueryTest.java

```diff
@@ -0,0 +1,540 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.calcite.avatica.SqlType;
+import org.apache.calcite.util.TimestampString;
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.granularity.Granularities;
+import org.apache.druid.query.Druids;
+import org.apache.druid.query.aggregation.CountAggregatorFactory;
+import org.apache.druid.query.aggregation.DoubleSumAggregatorFactory;
+import org.apache.druid.query.aggregation.FilteredAggregatorFactory;
+import org.apache.druid.query.dimension.DefaultDimensionSpec;
+import org.apache.druid.query.groupby.GroupByQuery;
+import org.apache.druid.query.ordering.StringComparators;
+import org.apache.druid.query.scan.ScanQuery;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.sql.calcite.filtration.Filtration;
+import org.apache.druid.sql.calcite.util.CalciteTests;
+import org.apache.druid.sql.http.SqlParameter;
+import org.junit.Test;
+
+public class CalciteParameterQueryTest extends BaseCalciteQueryTest
```

Comment:
This file looks like a clone of `CalciteQueryTest` with parts of each query substituted to have dynamic parameters and can become hard to maintain as more tests are added to `CalciteQueryTest`. Perhaps a better testing strategy is to focus on the query behavior that is uniquely affected by having dynamic parameters. For example, tests for dynamic parameters of each of the allowable dynamic parameter types, different numbers of dynamic parameters, invalid dynamic parameter ordinal/type/value, etc.).

Reply:
Added a comment about this files lineage and clarify intention, as my intention with this file is what you suggested, to provide a selection of different types and positions for parameters to have reasonably decent coverage on possibilities. I have renamed all tests to reflect this intention, as well as filled in some additional missing areas.